src/main/java/com/pagsestagio/movieapi/model/Filme.java

```diff
@@ -1,4 +1,65 @@
 package com.pagsestagio.movieapi.model;
 
-public record Filme(Integer identificador, String nomeFilme) {
-}
+import jakarta.persistence.*;
+
+import java.util.Objects;
+import java.util.UUID;
+
+
+@Entity
+public class Filme {
+    public Filme(Integer id, UUID idPublico, String nome) {
+        this.id = id;
+        this.idPublico = UUID.randomUUID();
+        this.nome = nome;
+    }
+
+    public Filme() {
+        this.idPublico = UUID.randomUUID();
+    }
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Integer id;
+
+    private UUID idPublico;
+
+    private String nome;
+
+    public Integer getId() {
+        return id;
+    }
+
+    public void setId(Integer id) {
+        this.id = id;
+    }
+
+    public UUID getIdPublico() {
+        return idPublico;
+    }
+
+    public void setIdPublico(UUID idPublico) {
+        this.idPublico = idPublico;
+    }
+
+    public String getNome() {
+        return nome;
+    }
+
+    public void setNome(String nome) {
+        this.nome = nome;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Filme filme = (Filme) o;
+        return Objects.equals(id, filme.id) && Objects.equals(idPublico, filme.idPublico) && Objects.equals(nome, filme.nome);
```

Comment:
sabe o motivo de ter escrito o equals e o hascode?

Reply:
Já vi bastante o uso de equals em IFs e tal, mas o hashCode é algo mais abstrato pra mim, não entendo muito bem. Neste caso aí, eu segui até aquele vídeo que você tinha me mandado no início, o cara gerava o hash code, não sei se isso é algo obrigatório ou não, mas creio que isso tenha a ver com o banco de dados também, auxiliar ali na hora de realizar alguma operação. Enfim, não sei o motivo.

Reply:
O equals fala se uma classe é igual a outra classe.

por exemplo você tem a classe:

class Filme{
 String nome;
 Integer id;
    public Filme(String nome, Integer id) {
        this.nome = nome;
        this.id = id;
    }  
}

ai se você cria dois objeto filmes
var filme1 = new Filme("teste", 1000)
var filme2 = new Filme("teste", 1000)

e tenta comparar eles
filme1.equals(filme2) //  false
isso ai dá false.

por que?
porque toda classe java extende a classe Object
e tem por padrão a hash e equals.
![image](https://github.com/Jhots/movie-api/assets/14281397/dd545e65-e2b7-4a66-886f-bca92d76d197)

Então seu filme1 e seu filme2  
vão ter um hash diferentes. E o equals que não compara os campos.

Ai você vai lá e sobreescreve o equals.

public boolean equals(Filme other) {
     return this.id = other.id && this.nome == this.nome
}

quando você faz o 
filme1.equals(filme2) agora isso vai dar true, porque ele compara o campos.

mas porque o sobreescrever o hash?

porque o hash é tipo a digital de um objeto que é usando em hash.
imagina que você tá usando um Set. o set é um estrutura de dados do Java que não permite repetição e usa o hash;

se você fizer isso aqui;
Set<Integer> set = new HashSet<>();
set.add(1);
set.add(1);
e depois ver o tamanho
set.size() // 1
vai dar um. Porque o hash vai ser igual 

agora se você fizer 
Set<Filme> filmes = new HashSet<>();
filmes.add(new Filme("nome", 1));
filmes.add(new Filme("nome", 1));
e depois ver o tamanho
filmes.sizer(); // 2
vai dar dois, pois os dois hash não iguais.

e ae você tem um problema. porque o set que é uma estrutura de dados que não deveria permitir repetição, tem objetos repetidos. 

e ae você implementa o método hash
public int hash() {
 return this.id.hash() + this.nome.hash();
}

e agora os dois filmes vão ter o mesmo hash; e quando você fizer o set
Set<Filme> filmes = new HashSet<>();
filmes.add(new Filme("nome", 1));
filmes.add(new Filme("nome", 1));
e depois ver o tamanho
filmes.sizer(); // 1
vai dar um, pois os dois hash são iguais.

Reply:
eles falam assim. se você sobreescreveu o equals, você vai ter que sobreescrever o hash.